skyportal/models.py

```diff
@@ -459,6 +463,64 @@ def gal_lon_deg(self):
         coord = ap_coord.SkyCoord(self.ra, self.dec, unit="deg")
         return coord.galactic.l.deg
 
+    @property
+    def luminosity_distance(self):
+        """
+        The luminosity distance in Mpc, using either DM or distance data
+        in the altdata fields or using the cosmology/redshift
+
+        Return None if the redshift puts the source not within the Hubble flow
+        """
+
+        # there may be a non-redshift based measurement of distance
+        # for nearby sources
+        if self.altdata:
+            if self.altdata.get("dm") is not None:
+                # see eq (24) of https://ned.ipac.caltech.edu/level5/Hogg/Hogg7.html
+                return (
+                    (10 ** (float(self.altdata.get("dm")) / 5.0)) * 1e-5 * u.Mpc
+                ).value
+            if self.altdata.get("parallax") is not None:
+                if float(self.altdata.get("parallax")) > 0:
+                    # assume parallax in arcsec
+                    return (1e-6 * u.Mpc / float(self.altdata.get("parallax"))).value
+
+            if self.altdata.get("dist_kpc") is not None:
+                return (float(self.altdata.get("dist_kpc")) * 1e-3 * u.Mpc).value
+            if self.altdata.get("dist_Mpc") is not None:
+                return (float(self.altdata.get("dist_Mpc")) * u.Mpc).value
+            if self.altdata.get("dist_pc") is not None:
+                return (float(self.altdata.get("dist_pc")) * 1e-6 * u.Mpc).value
+            if self.altdata.get("dist_cm") is not None:
+                return (float(self.altdata.get("dist_cm")) * u.Mpc / 3.085e18).value
+
+        if self.redshift:
+            if self.redshift * 2.99e5 * u.km / u.s < 350 * u.km / u.s:
+                # stubbornly refuse to give a distance if the source
+                # is not in the Hubble flow
+                return None
+            return (cosmo.luminosity_distance(self.redshift)).value
```

Comment:
```suggestion
            return (cosmo.luminosity_distance(self.redshift)).to(u.Mpc).value
```

Reply:
the API for `luminosity_distance` says it will return in Mpc

Reply:
Done.